pkg/ebpf/debugfs.go

```diff
@@ -68,13 +79,27 @@ func getProbeStats(pid int, profile string) map[string]int64 {
 			event = parts[1]
 		}
 		event = strings.ToLower(event)
-		res[fmt.Sprintf("%s_hits", event)] = st.Hits
-		res[fmt.Sprintf("%s_misses", event)] = st.Misses
+		hitsKey := fmt.Sprintf("%s_hits", event)
+		missesKey := fmt.Sprintf("%s_misses", event)
+		gauges.Lock()
+		setOrCreateGauge(hitsKey, float64(st.Hits))
+		setOrCreateGauge(missesKey, float64(st.Misses))
+		gauges.Unlock()
+		res[hitsKey] = st.Hits
+		res[missesKey] = st.Misses
 	}
 
 	return res
 }
 
+// setOrCreateGauge checks the gauges map from the KprobeStats object for a Gauge. It creates the Gauge if it doesn't exist and sets it using val.
+func setOrCreateGauge(key string, val float64) {
+	if _, ok := gauges.g[key]; !ok {
+		gauges.g[key] = telemetry.NewGauge(kProbeTelemetryName, key, []string{}, fmt.Sprintf("Gauge tracking value of %s", key))
```

Comment:
Rather than separate metrics for each probe, I think we want to utilize tags. I think you want two metrics, `ebpf_kprobes.hits` and `ebpf_kprobes.misses` with a tag `name`. To accomplish this, I think you want something like this to create the gauges:
```
hitg := telemetry.NewGauge(kProbeTelemetryName, "hits", []string{"name"}, "Gauge tracking number of kprobe hits")
missg := telemetry.NewGauge(kProbeTelemetryName, "misses", []string{"name"}, "Gauge tracking number of kprobe misses")
```
and then in the `getProbeStats` function
```
hitg.Set(st.Hits, event)
missg.Set(st.Misses, event)
```

Reply:
If you want to get really fancy, having the `UID` as another tag would be awesome. We can leave that for a future PR though because it would require changing the parsing logic a bit.

Reply:
Separating kprobes from kretprobes (probably into separate metrics) is another improvement idea. We kind of already do this internally.

Reply:
Technically this code can be used to read `uprobe` data as well...